src/utils.cpp

```diff
@@ -332,8 +332,153 @@ void cnf_to_dimacs(Term cnf, std::ostringstream & y)
      y << 0;  // Symbolizing end of line
      y << "\n";
   }
+} 
+
+Term to_cnf(Term formula, SmtSolver s)
+{
+  
+  std::map<Term, Term>ma;//map ma contains the mapping of each term to it's new symbol in tseytin's transformation
+  
+  Sort boolsort = formula->get_sort();
+  assert(boolsort->get_sort_kind() == BOOL);
+  //vec stores the formulas which yet need to be given a symbolic name
+  TermVec vec = { formula };
+
+  //returning symbolic constants directly
+  if(formula->is_symbolic_const())
+  {
+    return formula;
+  }
+
+  static int pt = 1;// a pointer which will decide the next symbols name
+
+  auto term_gen = [&]
+  {
+    return (s->make_symbol("cnf__tseitin__fresh__var__" + std::to_string(pt++), boolsort));
+  };
+
+  auto give_symbolic_name=[&](Term t, bool push)
```

Comment:
I don't feel that strongly about this, but it might be a bit cleaner to have `give_symbolic_name` only handle names. For pushing onto the vector, you could either keep a separated `visited` cache, or just check the same conditions again separately. This is just something to consider.